QuEST/include/QuEST.h

```diff
@@ -600,6 +600,43 @@ void applyDiagonalOp(Qureg qureg, DiagonalOp op);
  */
 Complex calcExpecDiagonalOp(Qureg qureg, DiagonalOp op);
 
+/** Initialize an existing diagonal operator from an existing pauli hamiltonian.
+ * 
+ * @ingroup init
+ * @param[in,out] op the diagonal operator to initialize
+ * @param[in] hamil the pauli hamiltonian used to initialize op
+ * @param[out] offset, the constant element of the hamiltonian
```

Comment:
> Thanks very much for the PR! There are unfortunately a number of issues, possibly the most severe being the inconsistent use of identation, missing unit tests, and insufficient documentation. I've highlighted them below for your interest.
> 
> At this stage however, since I have to implement the parallel versions myself now, I don't think it is worth your time to make further edits, nor my time to review them. If you're happy, I will implement the parallel versions and documentation, and list you as the author of the "serial prototype"

Ok, I am happy with that. Just to make sure I understand it well, you don't expect me to solve the issues you pointed out, am I right?